hazelcast/src/main/java/com/hazelcast/internal/util/phonehome/PhoneHome.java

```diff
@@ -223,4 +238,12 @@ private void addClientInfo(PhoneHomeParameterCreator parameterCreator) {
                 .addParam("cpy", Integer.toString(clusterClientStats.getOrDefault(ConnectionType.PYTHON_CLIENT, 0)))
                 .addParam("cgo", Integer.toString(clusterClientStats.getOrDefault(ConnectionType.GO_CLIENT, 0)));
     }
+
+    private void addMapInfo(PhoneHomeParameterCreator parameterCreator) {
```

Comment:
Given that there will be several other metrics being collected, it would be better not to wrap all the metric collector calls into separate private methods of `PhoneHome`.

We can assume that there will be several further metric collector classes. Instead of using them in separate methods, it would be more concise if
 * there would be a `MetricCollector` interface
 * the `PhoneHome` would have a `List<MetricCollector> collectors` field, with a hardcoded list as its value
 * the PhoneHome would iterate on the `collectors` list, use them to compute the metric values, and put the values into the `PhoneHomeParameterCreator` instance.

For now we can also assume that
 * a single MetricCollector implementation will compute multiple metrics
 * the MetricCollector implementations can do its job given a `Node`, no other data source is necessary for it.

Given these guidelines, can you design the `MetricCollector` interface?

As the following step, can you further simplify the `PhoneHome` by extracting the cluster info and build info related metrics into `MetricCollector` implementations?